src/Ocis.php

```diff
@@ -548,72 +564,75 @@ public function getNotifications(): array
         }
 
         $content = $response->getBody()->getContents();
-        $ocsResponse = json_decode($content, true);
-        if (!is_array($ocsResponse)) {
-            throw new InvalidResponseException(
-                'Could not decode notification response. Content: "' .  $content . '"'
-            );
-        }
-        if (
-            !isset($ocsResponse['ocs']) ||
-            !array_key_exists('data', $ocsResponse['ocs']) ||
-            !is_array($ocsResponse['ocs']['data']) &&
-            !is_null($ocsResponse['ocs']['data'])
-        ) {
+        /**
+         * @phpstan-var array{
+         *  'ocs':array{
+         *      'data': ?array{
+         *          int, array{
+         *              notification_id: ?mixed,
+         *              app?: ?string,
+         *              user?: ?string,
+         *              datetime?: ?string,
+         *              object_id?: ?string,
+         *              object_type?: ?string,
+         *              subject?: ?string,
+         *              subjectRich?: ?string,
+         *              message?: ?string,
+         *              messageRich?: ?string,
+         *              messageRichParameters?:array{int, mixed}
+         *          }
+         *      }
+         *  }
+         *} $ocsResponse
+         */
+        $ocsResponse = (array)json_decode($content, true);
```

Comment:
The (array) cast should work here, even if the result of `json_decode` was a scalar (string, int, boolean) or null. It will produce an  array with one entry. That will be fine to pass to `isNotificationResponseValid` which will examine the array and work out that it is not a valid response.